plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/action/TransportGetWlmRuleAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule.action;
+
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.rule.action.GetRuleRequest;
+import org.opensearch.rule.action.GetRuleResponse;
+import org.opensearch.rule.service.RulePersistenceService;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+/**
+ * Transport action to get workload management Rules
+ * @opensearch.experimental
+ */
+public class TransportGetWlmRuleAction extends HandledTransportAction<GetRuleRequest, GetRuleResponse> {
+
+    private final RulePersistenceService rulePersistenceService;
```

Comment:
Shouldn't these `Action` be generic and get from appropriate place depending on the feature type?

Reply:
For each `ActionType` there is a singleton instance binding which is created in `ActionModule` ref: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/ActionModule.java#L766, config point: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/ActionModule.java#L1094. Hence making this generic will not work with existing dependency injection framework.

Reply:
Action have been made generic by creating Registry to get the appropriate PersistenceService